viewing_party/party.py

```diff
@@ -1,23 +1,149 @@
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    # make friends_movie_list 
+    friends_movie_list = []
+    for dictionary in user_data["friends"]:
+        for movie in dictionary["watched"]:
+            if movie not in friends_movie_list:
+                friends_movie_list.append(movie)
+    # alternative way: friends_movie_lists = [d["watched"] for d in user_data["friends"]] 
+
+    # iterate over user_data["watched"] and add each unique movie to a new empty list
+    unique_watched_list = []
+    
+    for movie in user_data["watched"]:
+        if movie not in friends_movie_list and movie not in unique_watched_list:
+            unique_watched_list.append(movie)
+
+    # return unique_watched_list
+    return unique_watched_list      
+
+def get_friends_unique_watched(user_data):
+    # make friends_movie_list 
+    friends_movie_list = []
+    for dictionary in user_data["friends"]:
+        for movie in dictionary["watched"]:
+            if movie not in friends_movie_list:
+                friends_movie_list.append(movie)
+    # alternative method: friends_movie_lists = [d["watched"] for d in user_data["friends"]]
+    #                     flat_list = [movie for sublist in friends_movie_lists for movie in sublist]
+    
+
+    # iterate over friends_movie_list and add each unique movie into a new empty list.
+    friends_unique_list = []
+    for movie in friends_movie_list:
+        if movie not in user_data["watched"] and movie not in friends_unique_list:
+            friends_unique_list.append(movie)
+    return friends_unique_list
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    # get friends_unique_watched_list
+    friends_unique_list = get_friends_unique_watched(user_data)
+    # iterate over friends unique list and see if each movie["host"] value \
+    # in user's subscription hosts
+    # if so, add to a recommended list
+    recommended_list = []
+    for movie in friends_unique_list:
+        if movie["host"] in user_data["subscriptions"]:
+            recommended_list.append(movie)
+    return recommended_list
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    friends_unique_list = get_friends_unique_watched(user_data)
+    genre_list = []
+    new_rec = []
+    for movie in user_data["watched"]:
+        genre_list.append(movie["genre"])
+    
+    if len(genre_list) > 0:  
+        max_genre = max(set(genre_list), key=genre_list.count) 
+        for movie in friends_unique_list:
+            if movie["genre"] == max_genre:
+                new_rec.append(movie)

```

Comment:
Very clever use of `max` and `set` here! As extra practice, can you consider how your might do this using a normal for loop and without a set?